src/dictionary/ui/display/WordDisplayParts.tsx

```diff
@@ -22,54 +23,163 @@ export function OtherForms({
   return (
     <>
       Other forms:{' '}
-      <ReactMarkdown renderers={{ paragraph: 'span' }}>
-        {forms.map((form) => otherForm(form)).join(', ')}
-      </ReactMarkdown>
+      <Markdown text={forms.map((form) => otherForm(form)).join(', ')} />
     </>
   )
 }
-interface Derivative {
+interface JoinProps {
+  list: unknown[]
+  separator: string
+  Component
+}
+export function Join({ list, separator, Component }: JoinProps): JSX.Element {
+  return (
+    <>
+      {' '}
+      {list.map((element, index) => (
+        <React.Fragment key={index}>
+          <Component {...element} />
```

Comment:
What's this? Why do you expand an element (which is actually `unknown` and not `Element`) to component props?

Reply:
The thing is i am passing a Component and a list which has as elements the props to the Component (which was passed) so the props can be anything. This is a litte clunky because if the element of the list dont have the exact type as the props of the Component (to which they are passed) it doesn't work. That's why i had to make a similiar function **JoinMarkdown**. I don't really know a better way currenlty to implmement a join function for ReactComponents-

Reply:
One way would be to get rid of the function and just use this everywhere  
```
<>
      {' '}
      {list.map((element, index) => (
        <React.Fragment key={index}>
          <Markdown text={element} />
          {index !== list.length - 1 ? separator : ''}
        </React.Fragment>
      ))}
    </>
```

Reply:
Please use descriptive names. I.e. do not call props element or list. 

Props cannot be anything. You can use generic types to enforce correct props.

Having a function is better than having duplicate joining everywhere. Would it work with elements instead of components? Could be possible to use `children` to pass the contents.